### PR TITLE
add pre-commit config file

### PR DESCRIPTION
**Pull request by JelleZijlstra**
Also gitignore vscode so my git status is clean.

The config file was copied from typeshed with fixes to some paths.

Refs #128

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
I don't think we want to blacken the test files for now, it moves the comments around too much.

**Comment by AlexWaygood** [comment]
Should we also remove the separate black and isort checks from the CI, like we did over at typeshed? (https://github.com/python/typeshed/pull/6376)

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,7 @@
 [tool.isort]
 profile = "black"
-combine_as_imports = true
\ No newline at end of file
+combine_as_imports = true
```
Comment (Akuli) [inline pyproject.toml:3]: I'm surprised that this isn't the default behavior.

**Comment by AlexWaygood** [comment]
We should also update the section on running tests in CONTRIBUTING, but that can be done in a separate PR.

File: pyi.py [hunk 2]
```diff
@@ -28,6 +28,8 @@
 
 if sys.version_info >= (3, 9):
     from ast import unparse
+
+
```
Comment (AlexWaygood) [inline pyi.py:32]: Huh, I was expecting these random extra two lines I added to be auto-reformatted by black.

**Comment by AlexWaygood** [comment]
Everything seems to be working now so I'm merging :) Thanks @JelleZijlstra!

**Pull request merged**